itest/lnd_onchain_test.go

```diff
@@ -731,3 +731,115 @@ func genAnchorSweep(ht *lntest.HarnessTest,
+	// wallet.
+	alice := ht.NewNode("Alice", nil)
+
+	const initialWalletAmt = btcutil.SatoshiPerBitcoin
+
+	// Funding the node with an initial balance.
+	ht.FundCoins(initialWalletAmt, alice)
+
+	// Create an address for Alice to send the coins to.
+	req := &lnrpc.NewAddressRequest{
+		Type: lnrpc.AddressType_WITNESS_PUBKEY_HASH,
+	}
+	resp := alice.RPC.NewAddress(req)
+
+	// We send half the amount to that address generating two unconfirmed
+	// outpoints in our internal wallet.
+	sendReq := &lnrpc.SendCoinsRequest{
+		Addr:   resp.Address,
+		Amount: initialWalletAmt / 2,
+	}
+	alice.RPC.SendCoins(sendReq)
+	txID := ht.Miner.AssertNumTxsInMempool(1)[0]
+
+	// Make sure the unspent number of utxos is 2 and the unconfirmed
+	// balances add up.
+	unconfirmed := ht.GetUTXOsUnconfirmed(
+		alice, lnwallet.DefaultAccountName,
+	)
+	require.Lenf(ht, unconfirmed, 2,
+		"expected 2 unconfirmed tx, got %v", len(unconfirmed))
+
+	var unconfirmedAmt btcutil.Amount
+	for _, utxo := range unconfirmed {
+		unconfirmedAmt += btcutil.Amount(utxo.AmountSat)
+	}
+
+	aliceBalResp := alice.RPC.WalletBalance()
+	require.Equal(ht, unconfirmedAmt,
+		btcutil.Amount(aliceBalResp.UnconfirmedBalance))
+
+	// Get the raw transaction to remove it later.
+	tx := ht.Miner.GetNumTxsFromMempool(1)[0]
+	// Calculate the tx fee so we can later compare the end amounts. We
+	// are sending from the internal wallet to the internal wallet so only
+	// the tx fee apply when calucalting the final amount of the wallet.
+	txFee := ht.CalculateTxFee(tx)
+
+	// Now remove the transaction. We should see that the wallet state
+	// equals the amount prior to sending the transaction. It is important
+	// to understand that we do not remove any transaction from the mempool
+	// (thats not possible in reality) we just remove it from our local
+	// store.
+	var buf bytes.Buffer
+	require.NoError(ht, tx.Serialize(&buf))
+	alice.RPC.RemoveTransaction(&walletrpc.Transaction{
+		TxHex: buf.Bytes(),
+	})
+
+	// Verify that the balance equals the initial state.
+	confirmed := ht.GetUTXOsConfirmed(
+		alice, lnwallet.DefaultAccountName,
+	)
+	require.Lenf(ht, confirmed, 1,
+		"expected 1 confirmed tx, got %v", len(confirmed))
+
+	var confirmedAmt btcutil.Amount
+	for _, utxo := range confirmed {
+		confirmedAmt += btcutil.Amount(utxo.AmountSat)
+	}
+
+	aliceBalResp = alice.RPC.WalletBalance()
+	require.Equal(ht, confirmedAmt,
+		btcutil.Amount(aliceBalResp.ConfirmedBalance))
+
+	// Mine a block and make sure the transaction previously broadcasted
+	// shows up in alice's wallet although we removed the transaction from
+	// the wallet when it was unconfirmed.
+	block := ht.Miner.MineBlocks(1)[0]
+	ht.Miner.AssertTxInBlock(block, txID)
+
+	// Verify that alice has 2 confirmed unspent utxos in her default
+	// wallet.
+	err := wait.NoError(func() error {
+		confirmed = ht.GetUTXOsConfirmed(
+			alice, lnwallet.DefaultAccountName,
+		)
+		require.Lenf(ht, confirmed, 2,
```

Comment:
need to replace `require` with `return fmt.Errorf(...)` tho otherwise it won't wait.